### PR TITLE
Refactor index handling on Matlab side

### LINKED ISSUE #150: Zero or one based index handling consolidation on the Matlab side
Matlab uses 1-based indexing opposed to 0 based indexing in C++. Currently we have a mixture of 0-based indexing and 1-based indexing in the Matlab side of the bindings, which should be unified to either one or the other.
- If we want to support 1-based indexing, the index manipulation and eventual error messages should be handled on the Matlab side.
- If we want to support 0-based indexing, we should add notes to all docstrings, since this deviates from the Matlab norm.

### PR DESCRIPTION
**Pull request by mpsonntag**
This PR refactors all Indexes on the Matlab-side to Matlab-style 1-based index; closes #150.

Successfully built and tested under win32 (Matlab R2011a) and win64 (Matlab R2011a, R2014b).

### CODE REVIEW

File: +nix/MultiTag.m [hunk 1]
```diff
@@ -52,17 +52,21 @@
...
         end
 
-        function r = open_reference_idx(obj, idx)
+        function r = open_reference_idx(obj, index)
+            idx = nix.Utils.handle_index(index);
             r = nix.Utils.open_entity(obj, 'openReferenceIdx', idx, @nix.DataArray);
         end
 
-        function r = retrieve_data(obj, pos_idx, id_or_name)
+        function r = retrieve_data(obj, position_idx, id_or_name)
```
Comment (jgrewe) [inline +nix/MultiTag.m]: if we spell out index in all other places, I would vote for doing the same here... other opinions?

Comment (mpsonntag) [reply +nix/MultiTag.m]: It was more along the lines of using `idx` in a compound word and `index` if its on its own, but I don't mind either way.

Comment (jgrewe) [reply +nix/MultiTag.m]: fine with me, @achilleas-k  any thoughts?

Comment (achilleas-k) [reply +nix/MultiTag.m]: Yeah, I think shortening index to idx when it's coupled with another word is OK, even though we use the full word when it's on its own.

**Pull request merged**